src/t_zset.c

```diff
@@ -1510,25 +1527,38 @@ long zsetRank(robj *zobj, sds ele, int reverse) {
     if (zobj->encoding == OBJ_ENCODING_LISTPACK) {
         unsigned char *zl = zobj->ptr;
         unsigned char *eptr, *sptr;
+        unsigned long index = 1;
+        double curr, prev = 0;
 
-        eptr = lpSeek(zl,0);
+        if (!reverse) {
+            eptr = lpSeek(zl,0);
+            sptr = lpNext(zl,eptr);
+        } else {
+            sptr = lpSeek(zl,-1);
+            eptr = lpPrev(zl,sptr);
+        }
```

Comment:
```suggestion
        eptr = lpSeek(zl,reverse ? -2 : 0);
        sptr = lpNext(zl,eptr);
```
Would it be better?

Reply:
Thanks. I think it is better. This will only change one line.